components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/Extensions/CertificateStoreEx.cs

```diff
@@ -4,45 +4,83 @@
     /// Certificate store extensions
     /// </summary>
     public static class CertificateStoreEx {
 
+
         /// <summary>
-        /// Create trust list
+        /// Build the certificates trust list
         /// </summary>
-        /// <param name="certificateStore"></param>
-        /// <returns></returns>
-        public static CertificateTrustList ToCertificateTrustList(this CertificateStore certificateStore) {
-            var certificateTrustList = new CertificateTrustList {
-                StoreType = certificateStore.StoreType,
-                StorePath = certificateStore.StorePath
-            };
+        public static CertificateTrustList BuildCertificateTrustList(
+            this CertificateStore certificateStore,
+            CertificateTrustList certificateTrustList) {
+            if (certificateStore == null) {
+                throw new ArgumentNullException(nameof(certificateStore));
+            }
+
+            if (certificateTrustList == null) {
+                throw new ArgumentNullException(nameof(certificateTrustList));
+            }
+
+            if (certificateTrustList.StorePath != certificateStore.StorePath) {
+                certificateTrustList.StoreType = certificateStore.StoreType;
+                certificateTrustList.StorePath = certificateStore.StorePath;
+            }
 
             return certificateTrustList;
         }
 
         /// <summary>
-        /// Create identifier
+        /// Build the certificate identifier for the own app
         /// </summary>
-        /// <param name="certificateInfo"></param>
-        /// <param name="hostname"></param>
+        /// <param name="certificateStore"></param>
+        /// <param name="certificateIdentifier"></param>
         /// <returns></returns>
-        public static CertificateIdentifier ToCertificateIdentifier(
-            this CertificateInfo certificateInfo, string hostname) {
-            var certificateIdentifier = new CertificateIdentifier {
-                StoreType = certificateInfo.StoreType,
-                StorePath = certificateInfo.StorePath,
-                SubjectName = certificateInfo.SubjectName.Replace("DC=localhost", $"DC={hostname}")
-            };
+        public static CertificateIdentifier BuildCertificateIdentifier(
+            this CertificateInfo certificateStore,
+            CertificateIdentifier certificateIdentifier) {
+            if (certificateStore == null) {
+                throw new ArgumentNullException(nameof(certificateStore));
+            }
+
+            if (certificateIdentifier == null) {
+                throw new ArgumentNullException(nameof(certificateIdentifier));
+            }
+
+            if (certificateIdentifier.StorePath != certificateStore.StorePath) {
+                certificateIdentifier.StoreType = certificateStore.StoreType;
+                certificateIdentifier.StorePath = certificateStore.StorePath;
+            }
+
             return certificateIdentifier;
```

Comment:
Same comment as for other methods in this class, return values does not seem to be used.

Reply:
will refactor soon